### DISCUSSION TITLE
When running the same test in different suites, the test does not merge in report

### DISCUSSION THREAD

**Discussion by Awesomefreeman** [allure-framework/allure2#2132] [Questions & Support]
### Describe the Bug

When running the same test across different suites, test results are not combined in the report.

### Steps to Reproduce

1. Clone my exaple repo https://github.com/Awesomefreeman/MyAllureProject
2. Run SuiteOne
3. Run SuiteTwo
4. Generate Allure Report

### Expected Behaviour

I expect that both runs of the same test will be combined into one and I will see retries in Retries tab. 
I'll try to explain why this is needed. Let's imagine that I run 10 tests using the suite and 3 of the tests fails. I have a list of failed tests with a fully qualified name, then I want to retry these tests. I run the command `mvn clean -Dtest=ClassName#failed_test,... test`
I generate a report and expect to see 10 tests in the report, 3 of which are re-run. Now I will see 13 tests

Merge described in detail in this article https://habr.com/ru/companies/jugru/articles/337386/
![Screenshot 2023-09-29 at 15 29 34](https://github.com/allure-framework/allure2/assets/40743546/042ffd76-f4b9-42c9-9099-de1fbbeb244b)

### Screenshots or Additional Context

<img width="1497" alt="Screenshot 2023-09-29 at 15 22 45" src="https://github.com/allure-framework/allure2/assets/40743546/20ea54a2-9be1-4cc7-9d42-d9e4972a4eec">

### What Language are you using?

Java

### What Framework/Allure Integration you are using?

allure-junit5

### What version of Allure Integration you are using?

2.24.0

### What version of Allure Report you are using?

2.24.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by baev** [comment]
https://github.com/Awesomefreeman/MyAllureProject isn't available (status code 404)

**Reply by Awesomefreeman** [reply]
Change visibility option, pls try it again https://github.com/Awesomefreeman/MyAllureProject
